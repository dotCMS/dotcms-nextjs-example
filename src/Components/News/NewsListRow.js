import React from "react";
import PropTypes from "prop-types";
import { Media } from "reactstrap";

const NewsListRow = ({ news, configuration }) => {
  const displayedFields = configuration.fieldsToDisplay.split(",");
  return (
    <>
      <Media className="news-list-row" tag="li">
        {displayedFields.includes("image") ? (
          <Media left href={`/news-events/news/${news.identifier}`}>
            <Media object src={news.image} alt="image" />
          </Media>
        ) : (
          ""
        )}
        <Media body>
          {displayedFields.includes("title") ? (
            <Media heading>
              <a href={`/news-events/news/${news.identifier}`}>{news.title}</a>
            </Media>
          ) : (
            ""
          )}
          <h5>{displayedFields.includes("publishDate") ? news.sysPublishDate: ""}</h5>
          {displayedFields.includes("summary") ? news.lead : ""}
        </Media>
      </Media>
    </>
  );
};

NewsListRow.propTypes = {
    configuration: PropTypes.object,
    news: PropTypes.object.isRequired
};

export default NewsListRow;
