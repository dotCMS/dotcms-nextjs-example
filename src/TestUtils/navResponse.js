export const NavResponse = {
    entity: {
        children: [
            {
                href: '/about',
                title: 'About Us',
                folder: '123',
                type: 'htmlpage',
                children: []
            },
            {
                href: '/products',
                title: 'Products',
                folder: '456',
                type: 'htmlpage',
                children: []
            },
            {
                href: '/resources',
                title: 'Resources',
                folder: '789',
                type: 'folder',
                children: [
                    {
                        href: '/videos',
                        title: 'Videos',
                        type: 'htmlpage',
                        folder: null
                    },
                    {
                        href: '/blog',
                        title: 'Blog',
                        type: 'htmlpage',
                        folder: null
                    }
                ]
            }
        ]
    }
};
