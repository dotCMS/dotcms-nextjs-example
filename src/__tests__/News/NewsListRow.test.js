import React from 'react';
import { shallow } from 'enzyme';
import NewsListRow from '../../Components/News/NewsListRow';
import toJSON from 'enzyme-to-json';

import { newsMock } from '../../TestUtils/news';

describe('<NewsListRow />', () => {
    it('renders news correctly', () => {
        const conf = {
            fieldsToDisplay: 'title,image,summary,publishDate,summary'
        };
        const wrapper = shallow((<NewsListRow news={newsMock} configuration={conf} />));
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('renders news, without "title"', () => {
        const conf = {
            fieldsToDisplay: 'image,summary,publishDate,summary'
        };
        const wrapper = shallow((<NewsListRow news={newsMock} configuration={conf} />));
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('renders news, without "image"', () => {
        const conf = {
            fieldsToDisplay: 'title,summary,publishDate,summary'
        };
        const wrapper = shallow((<NewsListRow news={newsMock} configuration={conf} />));
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('renders news, without "publishDate"', () => {
        const conf = {
            fieldsToDisplay: 'title,summary,image,summary'
        };
        const wrapper = shallow((<NewsListRow news={newsMock} configuration={conf} />));
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('renders news, without "summary"', () => {
        const conf = {
            fieldsToDisplay: 'title,summary,publishDate,image'
        };
        const wrapper = shallow((<NewsListRow news={newsMock} configuration={conf} />));
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

})
