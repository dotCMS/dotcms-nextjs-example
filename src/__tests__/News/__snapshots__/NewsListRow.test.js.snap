// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsListRow /> renders news correctly 1`] = `
<Fragment>
  <Media
    className="news-list-row"
    tag="li"
  >
    <Media
      href="/news-events/news/2943b5eb-9105-4dcf-a1c7-87a9d4dc92a6"
      left={true}
    >
      <Media
        alt="image"
        object={true}
        src="/dA/2943b5eb-9105-4dcf-a1c7-87a9d4dc92a6/image/gas-pump.jpg"
      />
    </Media>
    <Media
      body={true}
    >
      <Media
        heading={true}
      >
        <a
          href="/news-events/news/2943b5eb-9105-4dcf-a1c7-87a9d4dc92a6"
        >
          The Gas Price Rollercoaster
        </a>
      </Media>
      <h5>
        2011-06-29 11:00:00.0
      </h5>
      Historically, the price of gasoline has been tied to changes in the cost of crude oil; as crude prices rise, consumers can expect to pay more at the pump, and the reverse is also true.
    </Media>
  </Media>
</Fragment>
`;

exports[`<NewsListRow /> renders news, without "image" 1`] = `
<Fragment>
  <Media
    className="news-list-row"
    tag="li"
  >
    <Media
      body={true}
    >
      <Media
        heading={true}
      >
        <a
          href="/news-events/news/2943b5eb-9105-4dcf-a1c7-87a9d4dc92a6"
        >
          The Gas Price Rollercoaster
        </a>
      </Media>
      <h5>
        2011-06-29 11:00:00.0
      </h5>
      Historically, the price of gasoline has been tied to changes in the cost of crude oil; as crude prices rise, consumers can expect to pay more at the pump, and the reverse is also true.
    </Media>
  </Media>
</Fragment>
`;

exports[`<NewsListRow /> renders news, without "publishDate" 1`] = `
<Fragment>
  <Media
    className="news-list-row"
    tag="li"
  >
    <Media
      href="/news-events/news/2943b5eb-9105-4dcf-a1c7-87a9d4dc92a6"
      left={true}
    >
      <Media
        alt="image"
        object={true}
        src="/dA/2943b5eb-9105-4dcf-a1c7-87a9d4dc92a6/image/gas-pump.jpg"
      />
    </Media>
    <Media
      body={true}
    >
      <Media
        heading={true}
      >
        <a
          href="/news-events/news/2943b5eb-9105-4dcf-a1c7-87a9d4dc92a6"
        >
          The Gas Price Rollercoaster
        </a>
      </Media>
      <h5 />
      Historically, the price of gasoline has been tied to changes in the cost of crude oil; as crude prices rise, consumers can expect to pay more at the pump, and the reverse is also true.
    </Media>
  </Media>
</Fragment>
`;

exports[`<NewsListRow /> renders news, without "summary" 1`] = `
<Fragment>
  <Media
    className="news-list-row"
    tag="li"
  >
    <Media
      href="/news-events/news/2943b5eb-9105-4dcf-a1c7-87a9d4dc92a6"
      left={true}
    >
      <Media
        alt="image"
        object={true}
        src="/dA/2943b5eb-9105-4dcf-a1c7-87a9d4dc92a6/image/gas-pump.jpg"
      />
    </Media>
    <Media
      body={true}
    >
      <Media
        heading={true}
      >
        <a
          href="/news-events/news/2943b5eb-9105-4dcf-a1c7-87a9d4dc92a6"
        >
          The Gas Price Rollercoaster
        </a>
      </Media>
      <h5>
        2011-06-29 11:00:00.0
      </h5>
      Historically, the price of gasoline has been tied to changes in the cost of crude oil; as crude prices rise, consumers can expect to pay more at the pump, and the reverse is also true.
    </Media>
  </Media>
</Fragment>
`;

exports[`<NewsListRow /> renders news, without "title" 1`] = `
<Fragment>
  <Media
    className="news-list-row"
    tag="li"
  >
    <Media
      href="/news-events/news/2943b5eb-9105-4dcf-a1c7-87a9d4dc92a6"
      left={true}
    >
      <Media
        alt="image"
        object={true}
        src="/dA/2943b5eb-9105-4dcf-a1c7-87a9d4dc92a6/image/gas-pump.jpg"
      />
    </Media>
    <Media
      body={true}
    >
      <h5>
        2011-06-29 11:00:00.0
      </h5>
      Historically, the price of gasoline has been tied to changes in the cost of crude oil; as crude prices rise, consumers can expect to pay more at the pump, and the reverse is also true.
    </Media>
  </Media>
</Fragment>
`;
